### DISCUSSION TITLE
Add new language

### DISCUSSION THREAD

**Discussion by rolandow** [angularsen/UnitsNet#1170] [Q&A]
How can we add translations for an other language? Currently en-US and ru-RU seems to be defined, but how can we extend the library with our own language that does not exist in the library yet?

I assume we can define some json, and then load this to the library as well?

What would be the right procedure?

**Comment by angularsen** [comment]
Hi, you can simply add a new localization object to a unit's `Localization` array.

Here is Chinese (`zh-CN`), for example: 
https://github.com/angularsen/UnitsNet/blob/master/Common/UnitDefinitions/Length.json#L64-L77

This will have to be done for each unit you want to translate. The fallback is always en-US.

Then run `generate-code.bat` to update the source code.

**Reply by angularsen** [reply]
Oh, and in some cases you may have to also specify `AbbreviationsForPrefixes` if you need to override the default prefixes `k` for kilo, `M` for mega etc.

Example:
https://github.com/angularsen/UnitsNet/blob/9574d47ff4d0ac6ec49c1b9312ba7b60d47bb106/Common/UnitDefinitions/Energy.json#L141-L152
